modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/FormsRuleList.es.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {cleanup, render} from '@testing-library/react';
+import React from 'react';
+
+import {FormsRuleList} from '../../../../../src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/FormsRuleList.es';
+
+const defaultProps = () => {
+	return {
+		fields: [
+			{
+				fieldName: 'Text27344750',
+				label: 'Text27344750',
+			},
+			{
+				fieldName: 'Text27344751',
+				label: 'Text27344751',
+			},
+		],
+		keywords: '',
+		onDelete: () => {},
+		onEdit: () => {},
+		rules: [
+			{
+				actions: [
+					{
+						action: 'enable',
+						label: 'Text27344750',
+						target: 'Text27344750',
+					},
+				],
+				conditions: [],
+				logicalOperator: 'OR',
+				name: {
+					en_US: 'teste rulers',
+				},
+			},
+		],
+	};
+};
+
+describe('RuleList', () => {
+	afterEach(() => {
+		cleanup();
+		jest.restoreAllMocks();
+	});
+
+	describe('basic component render', () => {
+		it('mounts component on empty state without data', () => {
+			const {getByText} = render(<FormsRuleList />);
+
+			expect(
+				getByText(
+					'there-are-no-rules-yet-click-on-plus-icon-below-to-add-the-first'
+				)
+			).toBeTruthy();
+		});
+
+		it('mounts with operands type field', () => {
+			const props = defaultProps();
+
+			props.rules[0].conditions.push({
+				operands: [
+					{
+						label: 'Text',
+						repeatable: false,
+						source: 0,
+						type: 'field',
+						value: 'Text27344750',
+					},
+					{
+						type: 'string',
+						value: '123',
+					},
+				],
+				operator: 'equals-to',
+			});
+
+			const {getByText} = render(<FormsRuleList {...props} />);
+
+			expect(getByText('is-equal-to')).toBeTruthy();
+		});
+
+		it('mounts with two operands type field', () => {
```

Comment:
This text is not very clear, I put here the same explanation above because in most tests we will always have two operands this means that we will have a binary operator otherwise an operator with only the left side condition, which is just for `is-empty` or `not-is-empty`.